content/guides/bake/index.md

```diff
@@ -0,0 +1,499 @@
+> caches Go modules between builds, improving build performance by avoiding the
+> need to re-download dependencies. This shared cache ensures that the same
+> dependency set is available across build, test, and other stages.
+
+Now, running the `test` target with Bake will evaluate the unit tests for this
+project. If you want to verify that it works, you can make an arbitrary change
+to `main_test.go` to cause the test to fail.
+
+Next, to enable linting, add another target to the Bake file, named `lint`:
+
+```hcl
+target "lint" {
+  target = "lint"
+  output = ["type=cacheonly"]
+}
+```
+
+And in the Dockerfile, add the build stage. This stage will use the official
+`golangci-lint` image on Docker Hub.
+
+> [!TIP]
+> Because this stage relies on executing an external dependency, it's generally
+> a good idea to define the version you want to use as a build argument. This
+> lets you more easily manage version upgrades in the future by collocating
+> dependency versions to the beginning of the Dockerfile.
+
+```dockerfile {hl_lines=[2,"6-8"]}
+ARG GO_VERSION="1.23"
+ARG GOLANGCI_LINT_VERSION="1.61"
+
+#...
+
+FROM golangci/golangci-lint:v${GOLANGCI_LINT_VERSION}-alpine AS lint
+RUN --mount=target=.,rw \
+    golangci-lint run
+```
+
+Lastly, to enable running both tests simultaneously, you can use the `groups`
+construct in the Bake file. A group can specify multiple targets to run with a
+single invocation.
+
+```hcl
+group "validate" {
+  targets = ["test", "lint"]
+}
+```
+
+Now, running both tests is as simple as:
+
+```console
+$ docker buildx bake validate
+```
+
+## Building variants
+
+Sometimes you need to build more than one version of a program. The following
+example uses Bake to build separate "release" and "debug" variants of the
+program, using matrices. Using matrices lets you run parallel builds with
+different configurations, saving time and ensuring consistency.
+
+A matrix expands a single build into multiple builds, each representing a
+unique combination of matrix parameters. This means you can orchestrate Bake
+into building both the production and development build of your program in
+parallel, with minimal configuration changes.
+
+The example project for this guide is set up to use a build-time option to
+conditionally enable debug logging and tracing capabilities.
+
+- If you compile the program with `go build -tags="debug"`, the additional
+  logging and tracing capabilities are enabled (development mode).
+- If you build without the `debug` tag, the program is compiled with a default
+  logger (production mode).
+
+Update the Bake file by adding a matrix attribute which defines the variable
+combinations to build:
+
+```diff {title="docker-bake.hcl"}
+ target "default" {
++  matrix = {
++    mode = ["release", "debug"]
++  }
++  name = "image-${mode}"
+   target = "image"
+```
+
+The `matrix` attribute defines the variants to build ("release" and "debug").
```

Comment:
Link to matrix docs https://docs.docker.com/build/bake/matrices/? Maybe show what it would look like without matrix in a note callout?

Reply:
I've aded a link to the matrix targets page. I would rather not show what it would look like without `matrix`, too long 😅